### ISSUE TITLE
mhchem: support complex expressions

### ISSUE THREAD

**Issue by fabiospampinato** [notable/notable#522]
Some complex expressions aren't currently rendered properly, for instance the following:

```katex
\ce{ Zn^2+ <=>[\ce{+ 2OH-}][\ce{+ 2H+}]$\underset{\text{amphoteric hydroxide}}{\ce{Zn(OH)2 v}}$<=>C[+2OH-][{+ 2H+}]$\underset{\text{tetrahydroxozincate}}{\cf{[Zn(OH)4]^2-}}$ }
```

Should get rendered as:

![641675f64220bf680e0cc8e053cf7a9a](https://user-images.githubusercontent.com/1812093/53972254-ba1f6000-40fe-11e9-8b63-4df4f9a2ebad.png)

But it seems the `\cf` function/macro isn't defined, and there might be problems with expressions containing the `$` character.

**Comment by fabiospampinato** [comment]
The following expression is also problematic:

```
$\ce{NaOH(aq,$\infty$)}$
```

**Comment by fabiospampinato** [comment]
I've added a note about expressions containing `$`, they must be wrapped in double `$$`.

Regarding the complicated expression in the first this is an upstream bug: https://github.com/KaTeX/KaTeX/issues/2007

**Comment by fabiospampinato** [comment]
Fixed in v1.6.
